lib/pkgcloud/azure-v2/compute/client/servers.js

```diff
@@ -143,71 +149,110 @@ function createServer(options, callback) {
-
-    // Adding additional data disks
-    if (options.storageDataDiskNames && options.storageDataDiskNames.length) {
-      options.storageDataDiskNames.forEach(function (ddName, idx) {
-        template.resources[vmIndex].properties.storageProfile.dataDisks.push({
-          'name': 'datadisk' + idx.toString(),
-          'diskSizeGB': '100',
-          'lun': 0,
-          'vhd': {
-            'uri': '[concat(reference(concat(\'Microsoft.Storage/storageAccounts/\', variables(\'storageAccountName\')), \'2016-01-01\').primaryEndpoints.blob, parameters(\'storageContainerName\'),\'/\', \'' + ddName + '\', \'.vhd\')]'
-          },
-          'createOption': 'Empty'
-        });
-      });
+  var templateName = 'compute' + (options.imageSourceUri ? '-from-image' : '');
+  self.deploy(templateName, options, (err, result) => {
+
+    if (err) {
+      return callback(err);
+    }
+
+    var hostname = null;
+    var location = null;
+    var vmID = null;
+    if (result &&
+      result.properties &&
+      result.properties.outputs &&
+      result.properties.outputs.hostname &&
+      result.properties.outputs.hostname.value &&
+      result.properties.outputs.location &&
+      result.properties.outputs.location.value &&
+      result.properties.outputs.vmID &&
+      result.properties.outputs.vmID.value) {
+      hostname = result.properties.outputs.hostname.value;
+      location = result.properties.outputs.location.value;
+      vmID = result.properties.outputs.vmID.value;
+    } else {
+      return callback(new Error('Result was not in the correct format: ' + JSON.stringify(result || {})));
     }
 
-    // If this is a windows machine, add an extension that enables remote connection via WinRM
     if (options.osType === 'Windows') {
-      template.resources[vmIndex].resources = [{
-        'type': 'Microsoft.Compute/virtualMachines/extensions',
-        'name': '[concat(variables(\'vmName\'),\'/WinRMCustomScriptExtension\')]',
-        'apiVersion': constants.DEFAULT_API_VERSION,
-        'location': '[resourceGroup().location]',
-        'dependsOn': [
-          '[concat(\'Microsoft.Compute/virtualMachines/\', variables(\'vmName\'))]'
-        ],
-        'properties': {
-          'publisher': 'Microsoft.Compute',
-          'type': 'CustomScriptExtension',
-          'typeHandlerVersion': '1.4',
-          'settings': {
-            'fileUris': [
-              'https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/ConfigureWinRM.ps1',
-              'https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/makecert.exe',
-              'https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/winrmconf.cmd'
-            ],
-            'commandToExecute': '[concat(\'powershell -ExecutionPolicy Unrestricted -file ConfigureWinRM.ps1 \',variables(\'hostDNSNameScriptArgument\'))]'
```

Comment:
Right, sounds great, but this means two things:

1. Would be better to add this using `template.resources[vmIndex].resources = [{...`
2. For this to work E2E you'll need to change cloud-cd to only support ssh (for windows/linux) - but this is not in the scope of this repo